operator/builtin/input/goflow/goflow.go

```diff
@@ -0,0 +1,189 @@
+package goflow
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"sync"
+	"time"
+
+	flowmessage "github.com/cloudflare/goflow/v3/pb"
+	"github.com/cloudflare/goflow/v3/utils"
+	"github.com/jpillora/backoff"
+	"github.com/observiq/stanza/operator"
+	"github.com/observiq/stanza/operator/helper"
+	"go.uber.org/zap"
+)
+
+const (
+	operatorName     = "goflow_input"
+	modeSflow        = "sflow"
+	modeNetflowV5    = "netflow_v5"
+	modeNetflowIPFIX = "netflow_ipfix"
+)
+
+func init() {
+	operator.Register(operatorName, func() operator.Builder { return NewGoflowInputConfig("") })
+}
+
+// NewGoflowInputConfig creates a new goflow input config with default values
+func NewGoflowInputConfig(operatorID string) *GoflowInputConfig {
+	return &GoflowInputConfig{
+		InputConfig: helper.NewInputConfig(operatorID, operatorName),
+	}
+}
+
+// GoflowInputConfig is the configuration of a goflow input operator.
+type GoflowInputConfig struct {
+	helper.InputConfig `yaml:",inline"`
+
+	Mode    string `json:"mode,omitempty"    yaml:"mode,omitempty"`
+	Address string `json:"address,omitempty" yaml:"address,omitempty"`
+	Port    uint   `json:"port,omitempty"    yaml:"port,omitempty"`
+	Workers uint   `json:"workers,omitempty" yaml:"workers,omitempty"`
+}
+
+// Build will build a goflow input operator.
+func (c *GoflowInputConfig) Build(context operator.BuildContext) ([]operator.Operator, error) {
+	inputOperator, err := c.InputConfig.Build(context)
+	if err != nil {
+		return nil, err
+	}
+
+	switch c.Mode {
+	case "sflow", "netflow_v5", "netflow_ipfix":
+		break
+	default:
+		return nil, fmt.Errorf("%s is not a supported Goflow mode", c.Mode)
+	}
+
+	if c.Address == "" {
+		c.Address = "0.0.0.0"
+	}
+
+	if c.Port == 0 {
+		return nil, fmt.Errorf("port is a required parameter")
+	}
```

Comment:
Should any validation be done on these?

Reply:
Resolved. Switched to ListenAddress and added net.ResolveUDPAddr to validate.